sdk/src/main/java/com/hedera/hashgraph/sdk/RequestType.java

```diff
@@ -0,0 +1,436 @@
+     * Get Version
+     */
+    GET_VERSION_INFO(HederaFunctionality.GetVersionInfo),
+
+    /**
+     * Transaction Get Receipt
+     */
+    TRANSACTION_GET_RECEIPT(HederaFunctionality.TransactionGetReceipt),
+
+    /**
+     * Create Topic
+     */
+    CONSENSUS_CREATE_TOPIC(HederaFunctionality.ConsensusCreateTopic),
+
+    /**
+     * Update Topic
+     */
+    CONSENSUS_UPDATE_TOPIC(HederaFunctionality.ConsensusUpdateTopic),
+
+    /**
+     * Delete Topic
+     */
+    CONSENSUS_DELETE_TOPIC(HederaFunctionality.ConsensusDeleteTopic),
+
+    /**
+     * Get Topic information
+     */
+    CONSENSUS_GET_TOPIC_INFO(HederaFunctionality.ConsensusGetTopicInfo),
+
+    /**
+     * Submit message to topic
+     */
+    CONSENSUS_SUBMIT_MESSAGE(HederaFunctionality.ConsensusSubmitMessage),
+
+    UNCHECKED_SUBMIT(HederaFunctionality.UncheckedSubmit),
+    /**
+     * Create Token
+     */
+    TOKEN_CREATE(HederaFunctionality.TokenCreate),
+
+    /**
+     * Get Token information
+     */
+    TOKEN_GET_INFO(HederaFunctionality.TokenGetInfo),
+
+    /**
+     * Freeze Account
+     */
+    TOKEN_FREEZE_ACCOUNT(HederaFunctionality.TokenFreezeAccount),
+
+    /**
+     * Unfreeze Account
+     */
+    TOKEN_UNFREEZE_ACCOUNT(HederaFunctionality.TokenUnfreezeAccount),
+
+    /**
+     * Grant KYC to Account
+     */
+    TOKEN_GRANT_KYC_TO_ACCOUNT(HederaFunctionality.TokenGrantKycToAccount),
+
+    /**
+     * Revoke KYC from Account
+     */
+    TOKEN_REVOKE_KYC_FROM_ACCOUNT(HederaFunctionality.TokenRevokeKycFromAccount),
+
+    /**
+     * Delete Token
+     */
+    TOKEN_DELETE(HederaFunctionality.TokenDelete),
+
+    /**
+     * Update Token
+     */
+    TOKEN_UPDATE(HederaFunctionality.TokenUpdate),
+
+    /**
+     * Mint tokens to treasury
+     */
+    TOKEN_MINT(HederaFunctionality.TokenMint),
+
+    /**
+     * Burn tokens from treasury
+     */
+    TOKEN_BURN(HederaFunctionality.TokenBurn),
+
+    /**
+     * Wipe token amount from Account holder
+     */
+    TOKEN_ACCOUNT_WIPE(HederaFunctionality.TokenAccountWipe),
+
+    /**
+     * Associate tokens to an account
+     */
+    TOKEN_ASSOCIATE_TO_ACCOUNT(HederaFunctionality.TokenAssociateToAccount),
+
+    /**
+     * Dissociate tokens from an account
+     */
+    TOKEN_DISSOCIATE_FROM_ACCOUNT(HederaFunctionality.TokenDissociateFromAccount),
+
+    /**
+     * Create Scheduled Transaction
+     */
+    SCHEDULE_CREATE(HederaFunctionality.ScheduleCreate),
+
+    /**
+     * Delete Scheduled Transaction
+     */
+    SCHEDULE_DELETE(HederaFunctionality.ScheduleDelete),
+
+    /**
+     * Sign Scheduled Transaction
+     */
+    SCHEDULE_SIGN(HederaFunctionality.ScheduleSign),
+
+    /**
+     * Get Scheduled Transaction Information
+     */
+    SCHEDULE_GET_INFO(HederaFunctionality.ScheduleGetInfo);
+
+    final HederaFunctionality code;
+
+    RequestType(HederaFunctionality code) {
+        this.code = code;
+    }
+
+    static RequestType valueOf(HederaFunctionality code) {
```

Comment:
This definitely needs to be a switch; I know you mentioned it wasn't working, I'll poke it a bit and see what's up.

Reply:
Thanks